pkg/util/create_or_update.go

```diff
@@ -106,6 +106,16 @@ func CreateOrUpdate(client dynamic.ResourceInterface, obj *unstructured.Unstruct
 
 // CreateAnew creates a resource, first deleting an existing instance if one exists.
 func CreateAnew(client dynamic.ResourceInterface, obj runtime.Object) error {
+	return createAnewMayWait(client, obj, false)
+}
+
+// CreateAnewWait creates a resource, first deleting an existing instance if one exists,
+// waiting for the deletion to complete.
+func CreateAnewWait(client dynamic.ResourceInterface, obj runtime.Object) error {
+	return createAnewMayWait(client, obj, true)
+}
+
+func createAnewMayWait(client dynamic.ResourceInterface, obj runtime.Object, waitForDeletion bool) error {
```

Comment:
I think we can just call this `createAnew`.  In fact I suggest to just add the param to the public function  and keep it `CreateAnew` rather than adding a second function.  Also we can make this even more general by making the param `*metav1.DeleteOptions`,  like the k8s `Delete` method - than we wouldn't have to change the API again if the need for other delete options arises.